cfme/v2v/infrastructure_mapping.py

```diff
@@ -0,0 +1,305 @@
+        }
+    )
+
+    @property
+    def is_displayed(self):
+        return self.title.text == self.form_title_text
+
+
+class EditInfrastructureMappingView(InfrastructureMappingForm):
+    form_title_text = "Edit Infrastructure Mapping"
+    save = Button("Save")
+
+    @property
+    def is_displayed(self):
+        return self.title.text == self.form_title_text
+
+
+@attr.s
+class InfrastructureMapping(BaseEntity, Updateable):
+    """Class representing v2v infrastructure mappings
+    A mapping consists of datastore/network/cluster components
+    Each component is a list of source -> target mappings, which are many:many relationships
+    """
+
+    @attr.s
+    class InfraMappingComponent(object):
+        """A datastore/network/cluster mapping component
+        Modeling a many:many relationship of a single component mapping
+        """
+
+        comp_name = None
+
+        sources = attr.ib(validator=attr.validators.instance_of(list))
+        targets = attr.ib(validator=attr.validators.instance_of(list))
+
+        def fill_dict(self):
+            return {"source": self.sources, "target": self.targets}
+
+    @attr.s
+    class ClusterComponent(InfraMappingComponent):
+        comp_name = "cluster"
+
+    @attr.s
+    class DatastoreComponent(InfraMappingComponent):
+        comp_name = "datastore"
+
+    @attr.s
+    class NetworkComponent(InfraMappingComponent):
+        comp_name = "network"
+
+    name = attr.ib()
+    description = attr.ib()
+    clusters = attr.ib(validator=attr.validators.instance_of(list))
+    datastores = attr.ib(validator=attr.validators.instance_of(list))
+    networks = attr.ib(validator=attr.validators.instance_of(list))
+
+    def fill_dict(self):
+        """Generate a dictionary for filling the InfraMapping wizard
+        This is a dictionary which contains 3 lists, for the 3 components
+        Each component list has dictionary elements, that contain source + target mapping lists
+        Example:
+            {'cluster':
+                [
+                    {'source_cluster': ['test_source'], 'target_cluster': ['test_target']},
+                    {'source_cluster: ['source1', 'source2'], 'target_cluster: ['t1', 't2']
+                ]
+            }
+        Returns: dict, see above
+        """
+        return {
+            "general": {"name": self.name, "description": self.description},
+            "cluster": [component.fill_dict() for component in self.clusters],
+            "datastore": [component.fill_dict() for component in self.datastores],
+            "network": [component.fill_dict() for component in self.networks],
+        }
+
+    def update(self, updates):
+        """
+        Args:
+            updates: An entity instance, that we can call form_values on
+        Returns:
+        """
+        view = navigate_to(self, "Edit", wait_for_view=20)
+        # TODO recursive update of self.form_values and updates.form_values
```

Comment:
You can use riggerlib for this, should be pretty straightforward with the fill_dict methods implemented already.